src/Module/Photo.php

```diff
@@ -83,26 +83,41 @@ public static function rawContent(array $parameters = [])
 			}
 
 			if (!empty($parameters['nickname_ext'])) {
-				$nickname = pathinfo($parameters['nickname_ext'], PATHINFO_FILENAME);
-				$user = User::getByNickname($nickname, ['uid']);
-				if (empty($user)) {
-					throw new HTTPException\NotFoundException();
-				}
+				if (in_array($parameters['type'], ['contact', 'header'])) {
+					$guid = pathinfo($parameters['nickname_ext'], PATHINFO_FILENAME);
+					$account = DBA::selectFirst('account-user-view', ['id'], ['guid' => $guid], ['order' => ['uid' => true]]);
+					if (empty($account)) {
+						throw new HTTPException\NotFoundException();
+					}
+
+					$id = $account['id'];
+				} else {
+					$nickname = pathinfo($parameters['nickname_ext'], PATHINFO_FILENAME);
+					$user = User::getByNickname($nickname, ['uid']);
+					if (empty($user)) {
+						throw new HTTPException\NotFoundException();
+					}
 
-				$uid = $user['uid'];
+					$id = $user['uid'];
+				}
 			}
 
 			// User Id Fallback, to remove after version 2021.12
 			if (!empty($parameters['uid_ext'])) {
-				$uid = intval(pathinfo($parameters['uid_ext'], PATHINFO_FILENAME));
+				$id = intval(pathinfo($parameters['uid_ext'], PATHINFO_FILENAME));
 			}
 
 			// Please refactor this for the love of everything that's good
 			if (isset($parameters['id'])) {
-				$uid = $parameters['id'];
+				$id = $parameters['id'];
+			}
+
+			if (empty($id)) {
+				Logger::notice('No picture id was detected', ['parameters' => $parameters]);
```

Comment:
I suggest you log the full query string as well from `DI::args()->getQueryString`.